lua/copilot/init.lua

```diff
@@ -36,6 +40,19 @@ local config_handler = function(opts)
   return user_config
 end
 
+M.auth = function ()
```

Comment:
Once you put M.auth in utils, add this to create_cmds

```
  vim.api.nvim_create_user_command("CopilotAuth", function()
    require("copilot.util").auth()
  end, {})
```

Reply:
Might have a chicken-egg problem adding this to `create_cmds`, (assuming `:CopilotAuth` is supposed to be callable even before setup is done... thinking of a situation where setup is done lazily when user first enters insert mode or something).

Any reason for not adding these commands to `plugin/copilot.lua`?

Reply:
> Might have a chicken-egg problem adding this to `create_cmds`, (assuming `:CopilotAuth` is supposed to be callable even before setup is done... thinking of a situation where setup is done lazily when user first enters insert mode or something).

That's a good point. You could wrap it in its own function and call said function once things are ready.

> Any reason for not adding these commands to `plugin/copilot.lua`?
That's a good idea as long as it doesn't cause the chicken and egg problem you described. I was concerned about needing control over when the cmds were registered.

Reply:
Nevermind, kept it simple for now. `:CopilotAuth` will be used by the user only once. So I don't think we need to support lazy loading right now.

If somebody lazy loaded, they'll just have to run `:CopilotAuth` once it's defined after `require("copilot").setup(...)`